src/Scaffolding/VS.Web.CG.EFCore/EntityFrameworkModelProcessor.cs

```diff
@@ -554,10 +556,62 @@ private DbContext TryCreateContextUsingAppCode(Type dbContextType, Type startupT
             }
             catch (Exception ex)
             {
+                System.Diagnostics.Debugger.Launch();
+                var exceptionType = ex.GetType();
+                // if MySQL exception with error code 1045, discard error message since it contains sensitive dev information
+                if (exceptionType.Name.Equals(MySqlException, StringComparison.OrdinalIgnoreCase) || exceptionType.FullName.Contains(MySqlException, StringComparison.OrdinalIgnoreCase))
+                {
+                    if (ex.Data.Keys.Count > 0 &&
+                        ex.Data["Server Error Code"] != null &&
+                        //based on error code 1045 from here https://dev.mysql.com/doc/
+                        ex.Data["Server Error Code"].ToString().Equals("1045", StringComparison.OrdinalIgnoreCase))
+                    {
+                        //userName will have some value, or will be empty. We're ok with both.
+                        string userName = GetUsername(ex.Message);
+                        ex = new Exception($"{string.Format(MessageStrings.MySQLDbContextExceptionMssg, userName)}\n");
+                        throw ex;
+                    }
+                }
                 throw ex.Unwrap(_logger);
             }
         }
 
+        /// <summary>
+        /// return the username/user id from exception messages including sensitive information.
+        /// Currently accounting for exception messages using MySQL scenarios.
+        /// </summary>
+        /// <returns>username from connection strings found in exception messages.</returns>
+        internal static string GetUsername(string exceptionMssg)
+        {
+            string username = string.Empty;
+            if (!string.IsNullOrEmpty(exceptionMssg))
+            {
+                //if the exception message is in the format `Access denied for user 'admin'@'localhost',
+                //discard the entire message since username and machine name could be both sensitive based on
+                //our collected telemetry.
+                if (exceptionMssg.Contains("Access denied for user", StringComparison.OrdinalIgnoreCase))
```

Comment:
Does this work outside of English?

Reply:
No it does not. The tool itself has no localization support as well but have not tried using MySQL providers in different languages.